main.go

```diff
@@ -114,128 +240,107 @@ func main() {
 			thisName, _ := ioutil.ReadFile(namePath)
 			fmt.Printf("- %s\n", string(thisName))
 		}
-	case "attach":
-		if parentSessionID != "" {
-			refuseNesting()
+	case "detach":
+		if parentSessionID == "" {
+			fmt.Println("Must be within session to detach")
 			return
 		}
-
-		sessionName := os.Args[2]
-		sessionID, err := findSession(sessionName)
-		if err != nil {
-			fmt.Printf("Could not find session `%s`\n", sessionName)
-		}
-		attach(sessionID)
+		detach(parentSessionID)
 	default:
-		if parentSessionID == "" {
-			sessionName, isNew := defaultPrompt()
-			if sessionName == "" {
-				return
-			}
-
-			var sessionID string
-			if isNew {
-				sessionID = launchServer(sessionName)
-			} else {
-				var err error
-				sessionID, err = findSession(sessionName)
-				if err != nil {
-					fmt.Printf("Could not find session `%s`\n", sessionName)
-				}
-			}
-
-			attach(sessionID)
-		} else {
-			namePath := path.Join(threemuxDir, parentSessionID, "name")
-			thisName, _ := ioutil.ReadFile(namePath)
-			fmt.Printf("You're in session `%s`\n", string(thisName))
-			var choice string
-			for choice != "y" && choice != "n" {
-				fmt.Print("Would you like to detach? [Y/n] ")
-				fmt.Scanf("%s", &choice)
-				choice = strings.ToLower(choice)
-				if choice == "" {
-					choice = "y"
-				}
-			}
-			if choice == "y" {
-				detach(parentSessionID)
-			}
-		}
+		fmt.Println(helpText)
+		os.Exit(1)
 	}
 }
 
-func detach(parentSessionID string) {
-	net.Dial("unix", path.Join(threemuxDir, parentSessionID, "shutdown.sock"))
+type SessionInfo struct {
+	name           string
+	uuid           string
+	path           string
+	fdPath         string
+	killClientPath string
+	killServerPath string
+	detachPath     string
+	resizePath     string
+	logsPath       string
 }
 
-func findSession(sessionName string) (sessionID string, err error) {
-	os.MkdirAll(threemuxDir, 0666)
-	children, err := ioutil.ReadDir(threemuxDir)
-	if err != nil {
-		panic(err)
-	}
-
-	for _, child := range children {
-		id := child.Name()
-		namePath := path.Join(threemuxDir, id, "name")
-		nameRaw, _ := ioutil.ReadFile(namePath)
-		nameCleaned := strings.TrimSpace(string(nameRaw))
+func detach(parentSessionID string) {
+	net.Dial("unix", elaborateSessionInfo("", parentSessionID).detachPath)
+}
 
-		if nameCleaned == sessionName {
-			return id, nil
-		}
+func elaborateSessionInfo(name string, uuid string) SessionInfo {
```

Comment:
It might be useful to use pointers for this object (I know it's gonna be pointer hell), but it will save on memory usage since it looks like copies of this are being returned by a bunch of functions.

Reply:
True, although SessionInfo is only used during the boot/shutdown process. Perhaps I should make a separate PR that fixes unneeded pass-by-value throughout the repo, not just releated to SessionInfo?